src/main/java/com/incognia/api/clients/TokenAwareNetworkingClient.java

```diff
@@ -33,45 +35,67 @@ public <T, U> U doPost(
       String path, T body, Class<U> responseType, Map<String, String> queryParameters)
       throws IncogniaException {
     refreshTokenIfNeeded();
-    return networkingClient.doPost(
-        path,
-        body,
-        responseType,
-        Collections.singletonMap(AUTHORIZATION_HEADER, buildAuthorizationHeader()),
-        queryParameters);
+    Map<String, String> headers =
+        new HashMap<String, String>() {
+          {
+            put(USER_AGENT_HEADER, buildUserAgentHeader());
+            put(AUTHORIZATION_HEADER, buildAuthorizationHeader());
+          }
+        };
+    return networkingClient.doPost(path, body, responseType, headers, queryParameters);
   }
 
   public <T, U> U doPost(String path, T body, Class<U> responseType) throws IncogniaException {
     refreshTokenIfNeeded();
-    return networkingClient.doPost(
-        path,
-        body,
-        responseType,
-        Collections.singletonMap(AUTHORIZATION_HEADER, buildAuthorizationHeader()));
+    Map<String, String> headers =
+        new HashMap<String, String>() {
+          {
+            put(USER_AGENT_HEADER, buildUserAgentHeader());
+            put(AUTHORIZATION_HEADER, buildAuthorizationHeader());
+          }
+        };
+    return networkingClient.doPost(path, body, responseType, headers);
   }
 
   public <T> void doPost(String path, T body, Map<String, String> queryParameters)
       throws IncogniaException {
     refreshTokenIfNeeded();
-    networkingClient.doPost(
-        path,
-        body,
-        Collections.singletonMap(AUTHORIZATION_HEADER, buildAuthorizationHeader()),
-        queryParameters);
+    Map<String, String> headers =
+        new HashMap<String, String>() {
+          {
+            put(USER_AGENT_HEADER, buildUserAgentHeader());
+            put(AUTHORIZATION_HEADER, buildAuthorizationHeader());
+          }
+        };
+    networkingClient.doPost(path, body, headers, queryParameters);
   }
 
   public <T> T doGet(String path, Class<T> responseType) throws IncogniaException {
     refreshTokenIfNeeded();
-    return networkingClient.doGet(
-        path,
-        responseType,
-        Collections.singletonMap(AUTHORIZATION_HEADER, buildAuthorizationHeader()));
+    Map<String, String> headers =
+        new HashMap<String, String>() {
+          {
+            put(USER_AGENT_HEADER, buildUserAgentHeader());
+            put(AUTHORIZATION_HEADER, buildAuthorizationHeader());
+          }
+        };
+    return networkingClient.doGet(path, responseType, headers);
   }
 
   private String buildAuthorizationHeader() {
     return token.getTokenType() + " " + token.getAccessToken();
   }
 
+  private String buildUserAgentHeader() {
```

Comment:
i think this only needs to be built once, so this could be a static constant in the class i think